src/sidebar/components/TopBar.js

```diff
@@ -21,19 +21,21 @@ import UserMenu from './UserMenu';
 
 /**
  * @typedef TopBarProps
- * @prop {AuthState} [auth]
- * @prop {Bridge} bridge
- * @prop {boolean} [isSidebar] - Flag indicating whether the app is the sidebar or a top-level page.
- * @prop {() => any} [onLogin] - Callback invoked when user clicks "Login" button.
- * @prop {() => any} [onLogout] - Callback invoked when user clicks "Logout" action in account menu.
- * @prop {() => any} [onSignUp] - Callback invoked when user clicks "Sign up" button.
- * @prop {MergedConfig} [settings]
- * @prop {Object} [streamer]
+ * @prop {AuthState} auth
+ * @prop {Bridge} [bridge]
+ * @prop {boolean} isSidebar - Flag indicating whether the app is the sidebar or a top-level page.
+ * @prop {() => any} onLogin - Callback invoked when user clicks "Login" button.
+ * @prop {() => any} onLogout - Callback invoked when user clicks "Logout" action in account menu.
+ * @prop {() => any} onSignUp - Callback invoked when user clicks "Sign up" button.
+ * @prop {MergedConfig} settings
+ * @prop {import('../services/streamer').StreamerService} streamer
  */
 
 /**
  * The toolbar which appears at the top of the sidebar providing actions
  * to switch groups, view account information, sort/filter annotations etc.
+ *
+ * @param {TopBarProps} props
```

Comment:
Fix a mistake where the `TopBarProps` type was not actually used. This revealed various mistakes in the types.